### DISCUSSION TITLE
Show Number of Disabled Extensions in VS Code

### DISCUSSION THREAD

**Discussion by anasir975** [microsoft/vscode-discussions#86] [Extension Development QnA] [answered]
vscode.extension.all displays the number of extensions installed in the machine programmatically. Is there any way to find out the number or list of extensions that have been disabled programmatically?
I asked the same question in Stack Overflow and nobody replied yet :) .

**Comment by alefragnani** [comment, marked as answer]
Hi @anasir975 ,

If you loop through the enumerator, you can see a few attributes for each `Extension`, but unfortunately, there is no `enable` state. 

There is however an `isActive` state, but it only says if the extension _was activated_, which depends on `activationEvents` entry.

```ts
        for (const ext of vscode.extensions.all) {
            const anExtension = await vscode.extensions.getExtension(ext.id);
            if (!anExtension) { continue }

            const isActive = anExtension.isActive ? "$(check-all)" : "$(info)";

            const name = anExtension.packageJSON.displayName ?? anExt?.packageJSON.name;
            const publisher = anExtension.packageJSON.isBuiltin ? "(builtin)" : "";
            ...
        }
```

If you really need the `enabled/disabled` state, you will need to open a new issue on VS Code repo

Hope this helps

**Comment by anasir975** [comment]
I posted there as well but then they redirected me here

**Reply by gjsjohnmurray** [reply]
https://github.com/microsoft/vscode/issues/156802

It seemed worth checking here in case any other extension developers had come up with a way of doing this.

**Reply by alefragnani** [reply]
Ops.. maybe because you _asked a question_, and the recommended location for questions was Stack Overflow and Slack Channel, and now this Discussions repo.

So, the point is, if you really need the `enabled` state, instead of asking a question there, open a **Feature Request** issue (https://github.com/microsoft/vscode/issues/new?assignees=&labels=&template=feature_request.md), asking for a new API. Then, the VS Code team should evaluate it differently.

**Reply by anasir975** [reply]
Thank you so very much @gjsjohnmurray and @alefragnani :)

**Reply by alefragnani** [reply]
You're welcome!
